### PR TITLE
Video: add backgroundColor prop

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Add backgroundColor prop to fill the video's placeholder
Previously black color by default, now we can pass an alternative color: transparent.

This provides a fix for Box/Mask: Box/Mask containing Video shows unexpected corner border when hiding overflowing content
image

Fixes bugs:
- https://jira.pinadmin.com/browse/BUG-123359

BEFORE 
image
AFTER
image

BEFORE
image
AFTER
image

### Links

- [Jira](link to Jira ticket(s))
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
From @dangerismycat comments

How does this handle instances where we would show the bar? Does the bar no longer show? If so, is there a strategy to detect when the black background should be used vs transparent?
- The bar is shown but it’s transparent. I don’t think there’s a was to control if the video size matches the aspectRatio passed to the Video cmp. If they match, and no bg fill is shown, then transparent can be used without issues. I think we could document a little bit more when the filler shows, we have to rewrite the docs so I didn’t put much thoughts on adding new examples. This bug was previously closed as “Blending corners in videos is something that browsers don’t handle well and fixing it inside the library would require hacks. If this critical for your feature you might have to build a layer on top of video or some kind of custom component. ” as per Brent’s feedback. So this is a pretty straightforward solution that comes with caveats / tradeoffs (edited) 

It’s the “after” of the bunny video that worries me. The fix in Gestalt seems fine — it’s about how that is used in Pinboard. Moving from showing black bars to not showing them (transparent bars) is a big visual change that dramatically changes how we display videos, and could have big changes in user behavior (not to mention the design change). I think there are two options:
- Get signoff from Design (not sure who, but possibly designers on several teams since this would have wide-ranging impacts) to change to transparent bars, and possibly run an experiment to quantify user changes
- Find some way to reliably determine when the bars would be shown and when they would not, and use that to determine the backgroundColor for that Video

File: packages/gestalt/src/Video.js [hunk 1]
```diff
@@ -224,11 +227,13 @@ export default class Video extends PureComponent<Props, State> {
   };
 
   static defaultProps: {|
+    backgroundColor: 'black' | 'transparent',
```
Comment (dangerismycat) [inline packages/gestalt/src/Video.js:230]: nit: repeating this props isn't great, would be better to define once as its own type

**Review by dangerismycat — APPROVED** [review]

**Comment by christianvuerings** [comment]
@AlbertCarreras checking in - are we good to merge?

File: packages/gestalt/src/Video.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Video.js]: DONE

**Pull request merged**